### DISCUSSION TITLE
x-bind attribute with undefined value

### DISCUSSION THREAD

**Discussion by chrswk** [alpinejs/alpine#3921] [7. Bugs (archived)]
Not sure if this is a bug or how to "fix" it, but it's definitely unexpected:

```
<button x-bind:aria-current="PAGES.EXAMPLE === activePage ? 'page' : undefined">
```

This leaves an empty `aria-current` on the element when `activePage` does not match the constant. After reading up on [this pull request](https://github.com/alpinejs/alpine/pull/486) and the related issues, I didn't understand why this was happening. The code clearly states `[null, undefined, false]` and it also worked fine if I returned `null` or `false`:

https://github.com/alpinejs/alpine/blob/0effdaedbf403f732ece72977737974f82fd29a4/packages/alpinejs/src/utils/bind.js#L100

But it seems like the problem is here, where the value is set to a empty string. Everything matches in my case, since my expression contains a dot (but is not actually a "nested object"):

https://github.com/alpinejs/alpine/blob/0effdaedbf403f732ece72977737974f82fd29a4/packages/alpinejs/src/directives/x-bind.js#L33

-Chris

**Comment by ekwoka** [comment]
It's a bit of a tricky thing to handle...to preserve expected behaviors.

null should work though instead of undefined.

**Reply by chrswk** [reply]
Absolutely. I switched to:

```
<button x-bind:aria-current="PAGES.EXAMPLE === activePage && 'page'">
```

Do you mind explaining the reason why the result is set to empty string (line 33 in 'x-bind') in case of a "nested object"? Why not just leave it undefined?

**Reply by ekwoka** [reply]
The mention of nested object is off, but it's because you only get undefined mostly in cases of having the expression be accessing an object property.

It's mainly, I'm pretty sure, to prevent issues of `undefined` actually ending up as the attribute value.
